qiskit/providers/ibmq/ibmqjob.py

```diff
@@ -197,7 +211,7 @@ def _wait_for_result(self, timeout=None, wait=5):
             raise JobError(str(api_err))
 
         status = self.status()
-        if status is not JobStatus.DONE:
+        if not expect_error and status is not JobStatus.DONE:
```

Comment:
What about naming `expect_error` as `expect_job_error` or `expect_general_error`? it's more verbose, but can't be misinterpreted with: "I'm expecting errors, even though they are coming from each of the experiments individually"

Reply:
`expect_error` is no longer needed. I think this belongs to an outdated commit.